src/core/dfn-finder.js

```diff
@@ -0,0 +1,222 @@
+import { pub } from "./pubsubhub";
+import { wrapInner } from "./utils";
+
+const topLevelEntities = new Set([
+  "callback interface",
+  "callback",
+  "dictionary",
+  "enum",
+  "interface mixin",
+  "interface",
+  "typedef",
+]);
+
+// TODO: make these linkable somehow.
+// https://github.com/w3c/respec/issues/999
+// https://github.com/w3c/respec/issues/982
+const unlinkable = new Set(["maplike", "setlike", "stringifier"]);
+
+// This function looks for a <dfn> element whose title is 'name' and
+// that is "for" 'parent', which is the empty string when 'name'
+// refers to a top-level entity. For top-level entities, <dfn>
+// elements that inherit a non-empty [dfn-for] attribute are also
+// counted as matching.
+//
+// When a matching <dfn> is found, it's given <code> formatting,
+// marked as an IDL definition, and returned. If no <dfn> is found,
+// the function returns 'undefined'.
+export function findDfn(defn, parent, name, definitionMap, idlElem) {
+  return (
+    findAttributeDfn(defn, parent, name, definitionMap, idlElem) ||
```

Comment:
I think this really wants to be a `switch(defn.type)` statement.

Reply:
Both find(attr/op)Dfn wants to do a second try when no match, so the || chain. IMO a switch here will complicate the second try.

Reply:
Reading more, the current setup needlessly run `findNormalDfn()` twice because of the implicit chain. It seems it should be explicitly called when needed and then we can use `switch()` in simpler way.